core-bundle/src/Util/DatabaseUtil.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Util;
+
+use Doctrine\DBAL\Connection;
+
+class DatabaseUtil
+{
+    public function __construct(
+        private readonly Connection $connection,
+    ) {
+    }
+
+    public function quoteIdentifier(string $identifier): string
```

Comment:
We should not name this method `quoteIdentifier` IMO as this has caused confusion in the past. We also need to add a comment like we did in 285ba6b2ce9fcc1f43adcc70be9f742dea11007e

How about `quoteIdentifierUnsafe()`, `quoteIfIdentifier()`, `quoteIdentifierWithoutValidation()` or `quoteIdentifierWithoutEscaping()`


```suggestion
    /**
	 * Do not pass user input as $identifier to this method as only identifiers get
	 * quoted and SQL expressions get returned as is!
	 *
	 * @internal Do not use this method in your code; use the "quoteIdentifier()"
	 *           method of the "@database_connection" service instead
	 */
    public function quoteIdentifierUnsafe(string $identifier): string
```

Reply:
I don't like the `Unsafe`, it implies that the method does kinda not work. But it works exactly as intended for the 99% case we have in the back end DCA. I do agree the passing of user data is not intended, but the method quotes a database indentifier, not a value?

The `@internal` notice implies that you should basically do what this method does for you automatically. Check if the value is an identifier (regex), then check if there are multiple parts and then quote each part? All of this obviously is only necessary if you have an identifier that you do not know – otherwise the `@database` method is obviously a better choice.

Reply:
> The `@internal` notice implies that you should basically do what this method does for you automatically. Check if the value is an identifier (regex), then check if there are multiple parts and then quote each part?

No, you should not do that. You should quote the identifier before building an expression with it.

Reply:
yes, if you can. But if you are using a DCA configuration (like the core does), there's no way around this, is there? So maybe we should write that down, but not name the method "Unsafe"? Like `quoteDcaIdentifier`?